apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemResultMapper.java

```diff
@@ -0,0 +1,237 @@
+package com.comet.opik.domain;
+
+import com.comet.opik.api.DatasetItem;
+import com.comet.opik.api.DatasetItemSource;
+import com.comet.opik.api.ExperimentItem;
+import com.comet.opik.api.FeedbackScore;
+import com.comet.opik.api.ScoreSource;
+import com.comet.opik.utils.JsonUtils;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Sets;
+import io.r2dbc.spi.Result;
+import io.r2dbc.spi.Row;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.comet.opik.api.DatasetItem.DatasetItemPage.Column;
+import static com.comet.opik.api.DatasetItem.DatasetItemPage.Column.ColumnType;
+import static com.comet.opik.utils.ValidationUtils.CLICKHOUSE_FIXED_STRING_UUID_FIELD_NULL_VALUE;
+import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toMap;
+
+public class DatasetItemResultMapper {
```

Comment:
Minor: the class itself can have default visibility like the majority of the methods. Regarding the methods, for those only used internally better have them as private to start with. We can make them default later.